scalafix/rules/src/main/scala/migrate/ExplicitImplicitsRule.scala

```diff
@@ -0,0 +1,79 @@
+package migrate
+
+import scala.tools.nsc.reporters.StoreReporter
+import scala.util.Failure
+import scala.util.Success
+
+import scala.meta.Tree
+import scala.meta.internal.pc.ScalafixGlobal
+import scala.meta.internal.proxy.GlobalProxy
+
+import metaconfig.Configured
+import scalafix.patch.Patch
+import scalafix.v1._
+import utils.CompilerService
+import utils.ScalaExtensions._
+import utils.SyntheticHelper
+
+class ExplicitImplicitsRule(g: ScalafixGlobal) extends SemanticRule("ExplicitImplicits") {
+  override def description: String = "Show implicit parameters and conversions"
+
+  def this() = this(null)
+
+  override def withConfiguration(config: Configuration): Configured[Rule] =
+    if (config.scalacClasspath.isEmpty) {
+      Configured.error(s"config.scalacClasspath should not be empty")
+    } else {
+      val global = CompilerService.newGlobal(config.scalacClasspath, config.scalacOptions)
+      global match {
+        case Success(settings) =>
+          Configured.ok(new ExplicitImplicitsRule(new ScalafixGlobal(settings, new StoreReporter, Map())))
+        case Failure(exception) => Configured.error(exception.getMessage)
+      }
+    }
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    lazy implicit val unit: g.CompilationUnit = g.newCompilationUnit(doc.input.text, doc.input.syntax)
+
+    doc.synthetics.collect { case syn @ ApplyTree(function, arguments) =>
+      // case for implicit params
+      if (syn.toString.startsWith("*")) {
+        (for {
+          originalTree <- SyntheticHelper.getOriginalTree(syn)
+          args         <- getImplicitParams(originalTree)
+        } yield Patch.addRight(originalTree, "(" + args.mkString(", ") + ")")).getOrElse(Patch.empty)
+      } else if (syn.toString().contains("(*)")) {
+        Patch.empty
+      } else Patch.empty
+    }.toList.asPatch
+  }
+
+  def getImplicitParams(originalTree: Tree)(implicit unit: g.CompilationUnit): Option[List[String]] =
+    for {
+      term       <- SyntheticHelper.getTermName(originalTree)
+      context    <- CompilerService.getContext(term, g)
+      globalTree <- getTreeFromContext(context)
```

Comment:
The same sequence of calls is made in the `MigrationRule` class. I suggest extracting that to a new method in `CompilerService`:

```scala
def getTreeFromContext(term: Term.Name)(implicit g: ScalafixGlobal): Option[g.Tree] =
  getContext.flatMap(getTreeFromContext)
```